src/app/services/server-api.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class ServerApiService {
+  // TODO: Configure base Url, to point to hosted endpoint. (localhost:3333 isn't a valid endpoint.)
+  private baseUrl = 'https://jsonplaceholder.typicode.com/todos/1';  // URL to web api
+
+  constructor(private http: HttpClient) { }
+
+  // Base Function
+  public getFromDataBase(urlExtension: string): Observable<any> {
```

Comment:
I'd drop the Smurf Naming Convention and just call these `get`, `post`, `delete`.
I think it would be useful to have the query parameters (`?foo=bar,frob=baz`) represented as an object passed to this function (`{foo: 'bar', frob: 'baz'}`), so that callers don't need to know how to encode URLs

Reply:
Understood, How would I go about adding query parameters without knowing how many parameters are being passed in?

Reply:
You accept one argument, a Map or Object, and iterate over its keys. Use https://angular.io/guide/http#url-parameters , but don't accept an HttpParams argument; instead construct an HttpParams based on the object you receive.

Reply:
Still needs addressing